meerk40t/gui/scene/scenewidgets.py

```diff
@@ -668,6 +668,12 @@ def process_draw(self, gc):
             y0 = self.start_location[1]
             x1 = self.end_location[0]
             y1 = self.end_location[1]
+            # Determine Colour on selection mode: standard (from left top to right bottom) = Blue, else Green
```

Comment:
I am unclear about:
a. What the difference in selection modes is TL->BR from all three other directions?
b. Why we only have TL->BR and not 2 of the 4 directions?

Reply:
As far as I can see there is a different selection scheme: in leftup you establish whether a object to be selected neds to be fully encompassed by the selection rectangle (right_drag = sx <= ex and sy <= ey) or whether it suffices that the selection rectangle overlaps the boundary of the object (and i believe there is still a bug there)
The color selection just follows this logic

Reply:
Okay, fixed intersection check

Reply:
There are 4 possible directions to draw the rectangle - AFAICT, two do one type of selection, and two do the other.

Reply:
I have just tested this code and it seems actually to select differently to before. I think that the way it used to work was better so please revert.

Reply:
Yes, we treat selection going BR->TL differently. Rather than needing to select the entire outlines it only needs to cross the path of the bounding box.

Reply:
Tested in actual code. This actually works pretty well.